src/xrEngine/pure.h

```diff
@@ -98,9 +98,9 @@ class MessageRegistry
             messages[0].Object->OnPure();
         else
         {
-            for (const auto& message : messages)
-                if (message.Prio != REG_PRIORITY_INVALID)
-                    message.Object->OnPure();
+            for (int i = 0; i < messages.size(); i++)
```

Comment:
What's the reason for such 'downgrading' of the for-expression? As for me, the initial variant seems to be more cool.

Reply:
Using range-based for here with gsc allocator, I get a crash: `Exception thrown: read access violation.
message.Object was 0xF.`

Reply:
Во время цикла на 3 итерации, в вектор messages добавляется еще один элемент, из-за этого все и слетает. Я пока не выяснил почему так происходит.

Reply:
When you're adding some elements to the vector all iterators become invalid. That's why the crash happens.

This fix is correct.

We have faced with this problem with @intorr, when he edited renderers code 🙂

Reply:
Yes, all iterators became invalid. However, I think it is NOT a correct fix. Modifying vectors while iterating it is a very bad solution. It could lead to a bugs which could be very hard to find. Looks like the problem needs a further investigation.